packages/renderer/src/lib/dialogs/QuickPickInput.spec.ts

```diff
@@ -27,19 +27,18 @@ import { beforeAll, describe, expect, test, vi } from 'vitest';
 import type { InputBoxOptions, QuickPickOptions } from './quickpick-input';
 import QuickPickInput from './QuickPickInput.svelte';
 
-const sendShowQuickPickValuesMock = vi.fn();
-const sendShowInputBoxValueMock = vi.fn();
-const sendShowQuickPickOnSelectMock = vi.fn();
 const receiveFunctionMock = vi.fn();
+const sendShowQuickPickOnSelectMock = vi.fn().mockResolvedValue(undefined);
+const sendShowQuickPickValuesMock = vi.fn().mockResolvedValue(undefined);
 
 // mock some methods of the window object
 beforeAll(() => {
   (window.events as unknown) = {
     receive: receiveFunctionMock,
   };
-  (window as any).sendShowQuickPickValues = sendShowQuickPickValuesMock;
-  (window as any).sendShowInputBoxValue = sendShowInputBoxValueMock;
-  (window as any).sendShowQuickPickOnSelect = sendShowQuickPickOnSelectMock.mockResolvedValue(undefined);
+  // We use spyOn to ensure that the values are undefined / reset before tests.
+  vi.spyOn(window, 'sendShowQuickPickOnSelect').mockImplementation(sendShowQuickPickOnSelectMock);
+  vi.spyOn(window, 'sendShowQuickPickValues').mockImplementation(sendShowQuickPickValuesMock);
```

Comment:
what?

vi.resetAllMock should reset those mock? no need to use spyOn?

Reply:
Mock by default return undefined so why not removing that code ?

Reply:
Clear only sets the calls counts to zero, it does not change the implementation of the mock, you should not have to call `reset` after that

Reply:
I've tried and it still errors out. Imitation of vi test perhaps?

Reply:
I agree that spyOn should not be used there